x-pack/plugins/security_solution/public/management/components/artifact_entry_card/hooks/use_normalized_artifact.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+/* eslint-disable @typescript-eslint/naming-convention */
+
+import { useMemo } from 'react';
+import { ExceptionListItemSchema } from '@kbn/securitysolution-io-ts-list-types';
+import { AnyArtifact, ArtifactInfo } from '../types';
+import { TrustedApp } from '../../../../../common/endpoint/types';
+
+/**
+ * Takes in any artifact and return back a new data structure used internally with by the card's components
+ *
+ * @param item
+ */
+export const useNormalizedArtifact = (item: AnyArtifact): ArtifactInfo => {
+  return useMemo(() => {
+    const {
+      name,
+      created_by,
+      created_at,
+      updated_at,
+      updated_by,
+      description = '',
+      entries,
+    } = item;
+    return {
+      name,
+      created_by,
+      created_at,
+      updated_at,
+      updated_by,
+      description,
+      entries: (entries as unknown) as ArtifactInfo['entries'],
```

Comment:
Do we need to do something special for nested conditions? I know this is not needed on this iteration, just curious

Reply:
not yet. that is part of the work that will be needed once we work on adding this to Event Filters (and possibly exceptions).